lib/structures/Message.js

```diff
@@ -128,15 +128,28 @@ class Message extends Base {
     get cleanContent() {
         var cleanContent = this.content;
 
-        if(this.mentions) {
-            this.mentions.forEach((mention) => {
-                if(this.channel.guild) {
-                    var member = this.channel.guild.members.get(mention.id);
-                    if(member && member.nick) {
-                        cleanContent = cleanContent.replace(new RegExp(`<@!?${mention.id}>`, "g"), "@" + member.nick);
+        var emoteReg = new RegExp("<(:\\w+:)[0-9]+>", "g");
+        if(emoteReg.test(cleanContent)) {
+            cleanContent = cleanContent.replace(emoteReg, "$1");
+        }
+
+        var userMentionReg = new RegExp("<@!?([0-9]+)>", "g");
+        if(userMentionReg.test(cleanContent)) {
+            cleanContent = cleanContent.replace(userMentionReg, (matched, mentionId) => {
+                var mention = this.mentions && this.mentions.find(m => m.id === mentionId);
+                if(mention) {
+                    if(this.channel.guild) {
+                        var member = this.channel.guild.members.get(mentionId);
+                        if(member && member.nick) {
+                            return "@" + member.nick;
+                        }
                     }
+                    return "@" + mention.username;
+                } else if(this.author && (this.type === 6 || this.type === 7)) {
+                    return "@" + this.author.username;
```

Comment:
This only replaces with the mention's username. To work fully, it should instead test for a nick like the checks above. Most likely needs refactoring.